File: chall02/estina.py
#!/usr/bin/python3
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    estina.py                                          :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: estina <estina@student.42madrid.com>       +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2020/04/06 19:21:07 by estina            #+#    #+#              #
#    Updated: 2020/04/06 19:42:57 by estina           ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

import sys

CHAR_TO_MORSE = {
	" ": " ",
	"a": ".-",
	"b": "-...",
	"c": "-.-.",
	"d": "-..",
	"e": ".",
	"f": "..-.",
	"g": "--.",
	"h": "....",
	"i": "..",
	"j": ".---",
	"k": "-.-",
	"l": ".-..",
	"m": "--",
	"n": "-.",
	"o": "---",
	"p": ".--.",
	"q": "--.-",
	"r": ".-.",
	"s": "...",
	"t": "-",
	"u": "..-",
	"v": "...-",
	"w": ".--",
	"x": "-..-",
	"y": "-.--",
	"z": "--.."
}

if len(sys.argv) != 2 or not sys.argv[1]:
	exit("usage: " + sys.argv[0] + " <a-zA-Z string>")
string = ""
for c in sys.argv[1]:
	if c.lower() not in CHAR_TO_MORSE:
		exit("usage: " + sys.argv[0] + " <a-zA-Z string>")
	string += CHAR_TO_MORSE.get(c)
print(string)

File: chall05/estina.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   estina.c                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: estina <estina@student.42madrid.com>       +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2020/04/16 14:07:38 by estina            #+#    #+#             */
/*   Updated: 2020/04/16 14:59:57 by estina           ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

static char	ft_lower(char c)
{
	if (c >= 'A' && c <= 'Z')
		return (c + 32);
	else
		return (c);
}

static int	ft_check_c(char c, int type)
{
	if (c == 'c')
		return (type ? 1 : 0);
	return (type ? 0 : 1);

}

int			ft_ie_except_after_c(char* str)
{
	int		i;
	int		rule;

	if (!str || !*str)
		return (0);
	i = 0;
	rule = 2;
	while (str[i] && rule)
	{
		if (ft_lower(str[i]) == 'e' && ft_lower(str[i + 1]) == 'i' && i)
			rule = ft_check_c(ft_lower(str[i - 1]), 1);
		if (ft_lower(str[i]) == 'i' && ft_lower(str[i + 1]) == 'e')
			rule = i ? ft_check_c(ft_lower(str[i - 1]), 0) : 1;
		i++;
	}
	return (rule == 2 ? 1 : rule);
}
